### PR TITLE
Service changes for Leases in ES

### PR DESCRIPTION
**Pull request by kenoir**

### CODE REVIEW

File: leases/app/lib/LeaseStore.scala [hunk 1]
```diff
@@ -37,8 +39,8 @@ object LeaseStore {
   private def uuid = Some(UUID.randomUUID().toString)
   private def key(id: String) = UniqueKey(KeyEquals('id, id))
 
-  def put(lease: MediaLease) = Scanamo.put(client)(tableName)(lease.copy(id=uuid))
-  def delete(id: String) = Scanamo.delete(client)(tableName)(key(id))
+  def put(lease: MediaLease) = Future { Scanamo.put[MediaLease](client)(tableName)(lease.copy(id=uuid))  }
+  def delete(id: String) = Future { Scanamo.delete(client)(tableName)(key(id)) }
```
Comment (NickPapacostas) [inline leases/app/lib/LeaseStore.scala]: I added these so I could use the results as promises (i.e. chain subsequent actions), is this the right way to do it? 

I noticed the other methods in here returned mappable objects but these didn't so I wasn't sure.

File: media-api/app/lib/ImageExtras.scala [hunk 2]
```diff
@@ -24,10 +24,16 @@ object ImageExtras {
...
   def validityMap(image: Image): Map[String, Boolean] = Map(
-    "no_rights"           -> !hasRights(image.usageRights),
-    "missing_credit"              -> !hasCredit(image.metadata),
-    "missing_description"         -> !hasDescription(image.metadata)
+    "no_rights"            -> !hasRights(image.usageRights),
+    "missing_credit"       -> !hasCredit(image.metadata),
+    "missing_description"  -> !hasDescription(image.metadata)
+  )
+
+  def validityOverrides(image: Image): Map[String, Boolean] = Map(
```
Comment (NickPapacostas) [inline media-api/app/lib/ImageExtras.scala]: Open to other / better names/ways of doing this

File: media-api/app/lib/ImageResponse.scala [hunk 3]
```diff
@@ -159,32 +138,21 @@ object ImageResponse extends EditsResponse {
...
-    }
-
-    val baseLinksWithOptimised = optimisedPngLink match {
-      case Some(link) => link :: baseLinks
-      case None => baseLinks
+      List(optimisedLink, imageLink, usageLink, leasesLink)
     }
 
-    if (valid) (cropLink :: baseLinksWithOptimised) else baseLinksWithOptimised
+    if (valid) (cropLink :: baseLinks) else baseLinks
```
Comment (NickPapacostas) [inline media-api/app/lib/ImageResponse.scala]: I think these shouldn't be here

File: thrall/app/lib/ElasticSearch.scala [hunk 4]
```diff
@@ -117,7 +117,23 @@ object ElasticSearch extends ElasticSearchClient with ImageFields {
...
+    prepareImageUpdate(id){ request =>
+      request.setScriptParams( Map(
+        "leaseByMedia" -> asGroovy(leaseByMedia),
+        "lastModified" -> asGroovy(lastModified)
+      ).asJava)
+        .setScript(
+          replaceLeasesScript +
+            updateLastModifiedScript,
+          scriptType)
+      .executeAndLog(s"updating leases on image $id with: $leaseByMedia")
```
Comment (NickPapacostas) [inline thrall/app/lib/ElasticSearch.scala]: this is noisy (prints out the whole media lease) 

It was helpful for me but maybe not in the long term?

**Comment by NickPapacostas** [comment]
I think some of the optimised stuff is getting overridden here. 

Otherwise I just asked had a few conceptual comments, no blockers

File: leases/app/lib/LeaseStore.scala (hunk 1 above)
Comment (kenoir) [inline leases/app/lib/LeaseStore.scala]: I think Scanamo now supports async requests, but this is ok for now.

File: thrall/app/lib/ElasticSearch.scala (hunk 4 above)
Comment (kenoir) [inline thrall/app/lib/ElasticSearch.scala]: Probably fine for an audit trail to begin with.

**Comment by NickPapacostas** [comment]
👍

**Pull request merged**